### PR TITLE
More integration tests

### PR DESCRIPTION
**Pull request by aawsome**

### CODE REVIEW

File: crates/core/tests/integration.rs [hunk 1]
```diff
@@ -404,3 +405,62 @@ fn test_find(tar_gz_testdata: Result<TestSource>, set_up_repo: Result<RepoOpen>)
...
+    let paths = PathList::from_iter(Some(source.0.path().join("0/0/9/2")));
+    let _ = repo.backup(&opts, &paths, SnapshotFile::default())?;
+
+    // re-read index
+    let repo = repo.to_indexed_ids()?;
+    // third backup
+    let paths = PathList::from_iter(Some(source.0.path().join("0/0/9/3")));
+    let _ = repo.backup(&opts, &paths, SnapshotFile::default())?;
+
+    repo.delete_snapshots(&[snapshot1.id])?;
```
[the remark this answers is not in the document]
Comment (aawsome) [reply crates/core/tests/integration.rs:442]: This is the purpose of `prune`: Snapshots have been removed and now we want to purge the data contained in those snapshots. Here, we remove snapshot1 as the two follow-up snapshots share data with snapshot1. So we also test that the shared data is not removed by `prune`.

File: crates/core/tests/integration.rs [hunk 2]
```diff
@@ -404,3 +405,62 @@ fn test_find(tar_gz_testdata: Result<TestSource>, set_up_repo: Result<RepoOpen>)
...
+    let check_opts = CheckOptions::default().read_data(true);
+    repo.check(check_opts)?;
+
+    if !instant_delete {
+        // re-run if we only marked pack files. As keep-delete = 0, they should be removed here
+        let plan = repo.prune_plan(&prune_opts)?;
+        plan.do_prune(&repo, &prune_opts)?;
+        repo.check(check_opts)?;
+    }
+
```
[the remark this answers is not in the document]
Comment (aawsome) [reply crates/core/tests/integration.rs:464]: First, the most important thing is that `prune` (being the only command which removes pack or index files) doesn't leave the repository in an unsane condition. This is why `check` is called here. (to admit, check currently does not produce an error but only logs an `error!` - I'm about to improve this next).

And second, yes I'd like to add snapshot testing here. But I realized that no data structure does `impl Serialize` - so this has to wait until JSON support for the prune statistics has been added.

[the remark this answers is not in the document]
Comment (aawsome) [reply crates/core/tests/integration.rs:464]: And, as a reference, restic also only checks if `check` does not report any error in this kind of `prune`integration tests.

[the remark this answers is not in the document]
Comment (simonsan) [reply crates/core/tests/integration.rs:464]: Ok, I opened issues for the corresponding things so we can track them.

**Review by simonsan — APPROVED** [review]

**Pull request merged**